### PR TITLE
devour: init at 12

### PR DESCRIPTION
**Pull request by Mazurel**
###### Motivation for this change

Added devour, simple x11 tool that hides your current window before launching an external program.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/X11/devour/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchFromGitHub {
+    owner = "salman-abedin";
+    repo = "devour";
+    rev = version;
+    sha256 = "1qq5l6d0fn8azg7sj7a4m2jsmhlpswl5793clcxs1p34vy4wb2lp";
+  };
+
+  installPhase = ''
+    mkdir -p $out/local/usr/bin
+    mv devour $out/local/usr/bin
```
Comment (SuperSandro2000) [inline pkgs/tools/X11/devour/default.nix:14]:
```suggestion
    install -Dm755 devour $out/usr/bin
```

File: pkgs/tools/X11/devour/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  name = "devour-${version}";
+  src = fetchFromGitHub {
+    owner = "salman-abedin";
+    repo = "devour";
+    rev = version;
+    sha256 = "1qq5l6d0fn8azg7sj7a4m2jsmhlpswl5793clcxs1p34vy4wb2lp";
+  };
+
+  installPhase = ''
+    mkdir -p $out/local/usr/bin
```
Comment (SuperSandro2000) [inline pkgs/tools/X11/devour/default.nix:13]:
```suggestion
    mkdir -p $out/usr/bin
```

File: pkgs/tools/X11/devour/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, fetchFromGitHub, libX11 }:
+stdenv.mkDerivation rec {
+  version = "12";
+  name = "devour-${version}";
+  src = fetchFromGitHub {
+    owner = "salman-abedin";
+    repo = "devour";
+    rev = version;
+    sha256 = "1qq5l6d0fn8azg7sj7a4m2jsmhlpswl5793clcxs1p34vy4wb2lp";
+  };
```
Comment (SuperSandro2000) [inline pkgs/tools/X11/devour/default.nix:10]:
```suggestion
  pname = "devour";
  version = "12";


  src = fetchFromGitHub {
    owner = "salman-abedin";
    repo = "devour";
    rev = version;
    sha256 = "1qq5l6d0fn8azg7sj7a4m2jsmhlpswl5793clcxs1p34vy4wb2lp";
  };
```

File: pkgs/tools/X11/devour/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    mkdir -p $out/local/usr/bin
+    mv devour $out/local/usr/bin
+  '';
+
+  buildInputs = [ libX11 ];
+
+  meta = with stdenv.lib; {
+    description = "Devour hides your current window before launching an external program and unhides it after quitting.";
+    homepage = "https://github.com/salman-abedin/devour";
+    maintainers = [ maintainers.mazurel ];
```
Comment (SuperSandro2000) [inline pkgs/tools/X11/devour/default.nix:22]:
```suggestion
    maintainers = with maintainers; [ mazurel ];
```

File: pkgs/tools/X11/devour/default.nix (hunk 1 above)
Comment (peterhoeg) [reply pkgs/tools/X11/devour/default.nix]: That will give give you a devour executable named "bin" and we don't stick stuff in `usr`:

This is the right way:
`install -Dm555 -t $out/bin devour`

File: pkgs/tools/X11/devour/default.nix (hunk 2 above)
Comment (peterhoeg) [reply pkgs/tools/X11/devour/default.nix]: Again, now `usr`.

File: pkgs/tools/X11/devour/default.nix (hunk 1 above)
Comment (Mazurel) [reply pkgs/tools/X11/devour/default.nix]: I will change output to `$out/bin` and permissions to`555`, but I don't think that install command behaves like that. By default it copies compiled sources into destination, which is directory created by the command above (tested). Although I will change it anyway, as most of the other nix packages use `install` with `-t`.

Comment (peterhoeg) [reply pkgs/tools/X11/devour/default.nix]: Yep, that's much better!

**Comment by peterhoeg** [comment]
Please also split this into 2 commits - one with adding yourself to the maintainer list and the other being the devour package.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103227` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>devour</li>
  </ul>
</details>

File: pkgs/tools/X11/devour/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  };
+
+  installPhase = ''
+    install -Dm555 -t $out/bin devour
+  '';
+
+  buildInputs = [ libX11 ];
+
+  meta = with stdenv.lib; {
+    description = "Devour hides your current window before launching an external program and unhides it after quitting.";
```
Comment (IvarWithoutBones) [inline pkgs/tools/X11/devour/default.nix:20]: `meta.description` should not end with a period, and this seems a bit long.
```suggestion
    
    description = "Devour hides your current window when launching an external program";
    longDescription = "Devour hides your current window before launching an external program and unhides it after quitting";

File: pkgs/tools/X11/devour/default.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, fetchFromGitHub, libX11 }:
```
Comment (IvarWithoutBones) [inline pkgs/tools/X11/devour/default.nix:1]: Minor spacing thing:
```suggestion
{ stdenv, fetchFromGitHub, libX11 }:

```

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
A few suggestions :)

File: pkgs/tools/X11/devour/default.nix [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    sha256 = "1qq5l6d0fn8azg7sj7a4m2jsmhlpswl5793clcxs1p34vy4wb2lp";
+  };
+
+  installPhase = ''
+    install -Dm555 -t $out/bin devour
+  '';
+
+  buildInputs = [ libX11 ];
+
+  meta = with stdenv.lib; {
```
[the remark this answers is not in the document]
Comment (Mazurel) [reply pkgs/tools/X11/devour/default.nix:19]: There is no license for this repository, I have opened an issue but have got no response yet.

File: pkgs/tools/X11/devour/default.nix (hunk 4 above)
Comment (IvarWithoutBones) [reply pkgs/tools/X11/devour/default.nix]: Why is this a problem? The full maintainer list doesn't need to be in scope when there's just one entry, no? Looks a lot cleaner the way it was before IMO

**Review by IvarWithoutBones — APPROVED** [review]

File: pkgs/tools/X11/devour/default.nix (hunk 7 above)
[the remark this answers is not in the document]
Comment (berbiche) [reply pkgs/tools/X11/devour/default.nix:20]: Which means the package is currently unfree?

[the remark this answers is not in the document]
Comment (Mazurel) [reply pkgs/tools/X11/devour/default.nix:20]: The package used to have GNU GPL license (4 days ago), but the author removed LICENSE file in last commit so it is currently under no license (which means unfree ? I don't think you can change GNU GPL license like that). As I have stated above I am still waiting for the response from author.

[the remark this answers is not in the document]
Comment (Mazurel) [reply pkgs/tools/X11/devour/default.nix:20]: I have received a response from the author and set a proper license (GPLv2).

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103227` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>devour</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103227` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>devour</li>
  </ul>
</details>

**Pull request merged**